content/how-to/author-libraries.md

```diff
@@ -1,7 +1,207 @@
+...
+webpackNumbers.numtoword(3); // output is Three
+...
+
+// As a script tag
+
+<html>
+...
+<script src='https://unpkg.com/webpack-numbers' type='text/javascript'/>
+<script type='text/javascript'>
+    ...
+    /* webpackNumbers is available as a global variable */
+    webpackNumbers.wordtonum('Five') //output is 5
+    ...
+</script>
+</html>
+```
+
+For full library configuration and code please refer to [webpack-library-example](https://github.com/kalcifer/webpack-library-example)
+
+## Configure webpack
+
+Now the agenda is to bundle this library
+  - Without bundling lodash but requiring it to be loaded by the consumer.
+  - Name of the library is `webpack-numbers` and the variable is `webpackNumbers`.
+  - Library can be imported as `import webpackNumbers from 'webpack-numbers'` or `require('webpack-numbers')`.
+  - Library can be accessed through global variable `webpackNumbers` when included through `script` tag.
+  - Library can be accessed inside Node.js.
+
+### Add webpack
+
+Add basic webpack configuration.
+Also add a [`.babelrc`](https://babeljs.io/docs/usage/babelrc/) file so that you can write webpack config in es2015 syntax.
+
+__webpack.config.babel.js__
+```javascript
+export default () => (
+    {
+        entry: './index.js',
+        output: {
+            path: './dist',
+            filename: 'webpack-numbers.js'
+        }
+    }
+);
+
+```
+
+This adds basic configuration to bundle the library.
+
+### Add Loaders
+
+But it will not work without adding relevant loaders for transpiling the code.
+We will add a `babel-loader` as our library is written in es2015 syntax. Add preset [`babel-preset-es2015`](https://www.npmjs.com/package/babel-preset-es2015) to transpile es2015 to es5. But for webpack2 we can disable `modules` as webpack2 understands es2015 module system.
+Similarly a `json-loader` is required to precompile our json fixture file.
+
+```javascript
+export default () => (
+    {
+        ...
+        module: {
+            rules: [{
+                test: /.js$/,
+                exclude: /node_modules/,
+                use: 'babel-loader',
+                options: {
+                    presets: [
+                        ["es2015", { "modules": false }]
+                    ]
+                }
+            }, {
+                test: /.json$/,
+                exclude: /node_modules/,
+                use: 'json-loader'
+            }]
+        }
+    }
+);
+```
+### Add `externals`
+
+Now, if you run `webpack`, you will find that a largish bundle file is created. If you inspect the file, you will find that lodash has been bundled along with your code.
+It would be unnecessary for your library to bundle a library like `lodash`. Hence you would want to give up control of this external library to the consumer of your library.
+
+This can be done using the `externals` configuration as
+
+```javascript
+export.default () => ({
+    ...
+    externals: [{
+        '_' : 'lodash'
```

Comment:
@sokra - I did not understand what change is expected here? This is working in my example code.